src/Symfony/Bundle/FrameworkBundle/DependencyInjection/Compiler/ProfilerPass.php

```diff
@@ -37,11 +38,14 @@ public function process(ContainerBuilder $container)
             $priority = isset($attributes[0]['priority']) ? $attributes[0]['priority'] : 0;
             $template = null;
 
-            if (isset($attributes[0]['template'])) {
-                if (!isset($attributes[0]['id'])) {
+            $collectorClass = $container->findDefinition($id)->getClass();
+            $isTemplateAware = is_subclass_of($collectorClass, TemplateAwareDataCollectorInterface::class);
```

Comment:
Why not use `$collectorClass instanceof TemplateAwareDataCollectorInterface`?

Reply:
Since `$collectorClass` is a string the result of `$collectorClass instanceof TemplateAwareDataCollectorInterface` won't be as expected. Or maybe I miss something 🙂

Reply:
nvm, I missed the fact that is was a string.